api/genotypes_loader.api.inc

```diff
@@ -91,6 +109,110 @@ function genotypes_loader_helper_add_record_with_mode($record_type, $table, $mod
   }
 }
 
+/**
+ * Places records into a CSV file, and then checks if it is time to bulk load the file via a COPY command
+ *
+ * @param $record_count
+ *   The current count of records that are being loaded in.
+ * @param $record_type
+ *   A human-readable term for your record for error messages.
+ * @param $table
+ *   The chado table to insert into.
+ * @param $insert_values
+ *   An array of [table column] => [value] mappings that is specific to the insert.
+ * @param $file_name
+ *   An optional name for the file to write the values to
+ * @param $final_chunk
+ *   If set to true, then the remaining genotypes have been added to the file and it is time to copy them in.
+ * @return
+ *   TRUE if success (meaning, we reached the end of the function, not neccesarily that everything went swimmingly), FALSE if failed.
+ */
+
+function genotypes_loader_helper_copy_records($record_count, $record_type, $table, $insert_values = array(), $file_name = NULL, $final_chunk = FALSE) {
+
+  // The number of records we want to reach before copying them in all at once
+  $copy_chunk_size = 10000;
+
+  // If we weren't given the name of the file to copy from, then try to find it using the record_type and table name
+  // that we were given
+  if ($file_name === NULL) {
+    $file_stub = '/tmp/genotype_loader.';
+    $record_type = str_replace(' ', '_', $record_type);
+    $record_type = strtolower($record_type);
+    $file_name = $file_stub . $table . '-' . $record_type . '.csv';
+  }
```

Comment:
The generated filename should include the database you're copying into. Otherwise you will end up with collisions when loading a file on portal and testing the next one on clone. The connection information is in `global $databases;` so just dpm that to find the name.

Reply:
Done in commit https://github.com/UofS-Pulse-Binfo/genotypes_loader/commit/b0c01bd7b0e7c2faced8829a2089782b1fd9b160!